cmd/thanos/rule.go

```diff
@@ -748,8 +748,8 @@ func addDiscoveryGroups(g *run.Group, c *http_util.Client, interval time.Duratio
 
 	g.Add(func() error {
 		return runutil.Repeat(interval, ctx.Done(), func() error {
-			c.Resolve(ctx)
-			return nil
+			err := c.Resolve(ctx)
```

Comment:
Nice! No need for separate variable though.

Please see https://thanos.io/contributing/coding-style-guide.md/#avoid-defining-variables-used-only-once

Reply:
Thanks!